src/modules/notifications/notifications.service.ts

```diff
@@ -46,11 +49,15 @@ export class NotificationsService {
     for (const notification of pendingNotifications) {
       try {
         notification.deliveryStatus = DeliveryStatus.IN_PROGRESS;
+        await validateOrReject(plainToClass(CreateNotificationDataDto, notification.data), {
+          validationError: { target: false },
+        });
         await this.notificationQueueService.addNotificationToQueue(notification);
       } catch (error) {
         notification.deliveryStatus = DeliveryStatus.PENDING;
+        notification.result = { result: error };
         this.logger.error(`Error adding notification with id: ${notification.id} to queue`);
-        this.logger.error(JSON.stringify(error, ['message', 'stack'], 2));
+        this.logger.error(JSON.stringify(error, null, 2));
```

Comment:
Why this change?

Did you check if adding `null` is printing stack and message?

Reply:
When only `['message', 'stack']` is specified, the error is logged as follows:
![image](https://github.com/OsmosysSoftware/osmo-notify/assets/108812833/4ab0b7e7-0b18-45f5-a692-0eafdd83e40e)

`null` was used instead due to the way the `error` value itself is generated. It does not have a `message` or `stack` value:
![image](https://github.com/OsmosysSoftware/osmo-notify/assets/108812833/45e9faa3-e898-4199-8325-fb94de258d05)

And when using `null`, error is logged as follows:
![image](https://github.com/OsmosysSoftware/osmo-notify/assets/108812833/91e983ec-955e-48a0-aad8-990f2e27f533)

Reply:
Okay for now. But I thought it was done in that specific way for a reason before.